### ISSUE TITLE
On older kernels, file blacklisted in running jail can't be removed (from outside of jail)

### ISSUE THREAD

**Issue by manevich** [netblue30/firejail#261]
File blacklisted in running jail can't be removed (from outside of jail).
This causes serious inconvenience when using firejail with long running (eg. server) processes.
For example, preventing user from updating system normally, as files like `/bin/su`, `/bin/mount`, `/usr/bin/sudo` are blacklisted by default.

Steps to reproduce:
* OS: Debian Jessie, with stock kernel.
* Create file that will be blacklisted in jail, **file must be created before running firejail**:
```shell
$touch ~/.netrc
```
* Run firejail:
```shell
$firejail firefox
```
* From outside of jail, run
```shell
$rm ~/.netrc
rm: cannot remove ‘/home/user/.netrc’: Device or resource busy
```

**Comment by genodeftest** [comment]
I cannot reproduce this issue. Which version are you using? I am running version 59f6010 and cea5874 from git. Not with a user-local `~/.netrc` file nor with a global file like /usr/bin/su.

**Comment by manevich** [comment]
Sorry, my fault. Looks like I left some runaway firejail process from firejail-0.9.36-1 Debian package running while cheked this.
Rechecked it now with corresponding https://github.com/netblue30/firejail/commit/02aebde80f008e71736e92c3c538aec075d83635 and also can't reproduce.

Look like some bug introduced by Debian mantainers. Still loking into this.

**Comment by manevich** [comment]
Well, I was wrong closing this issue.
Reproduced it with https://github.com/netblue30/firejail/commit/cea58747d61dc56ff8bb57aa02786cd8cc423bca version.

File must be created before running firejail, missed this when closed bug.
I am using Debian Jessie with stock kernel:
```
$ cat /etc/debian_version
8.3
$ uname -a
Linux localhost 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt20-1+deb8u3 (2016-01-17) x86_64 GNU/Linux
```
Updating description with some clarifications.

**Comment by manevich** [comment]
Looks like issue with old kernel, on same system with newer kernel everything fine.
```
$ uname -a
Linux localhost 4.3.0-0.bpo.1-amd64 #1 SMP Debian 4.3.3-7~bpo8+1 (2016-01-19) x86_64 GNU/Linux
```

**Comment by genodeftest** [comment]
That's a good explanation. I was running a 4.3.x kernel when unable to reproduce.

**Comment by netblue30** [comment]
I'm surprised this didn't come earlier.  Blacklisting is implementing by mounting an empty, read-only file or directory on top of the original file. The kernel, at least the older kernels, will refuse to delete the file because it is a mount point in some other place in the system. There is nothing we can do about it at application level.

**Comment by netblue30** [comment]
I documented the problem on project page: https://firejail.wordpress.com/support/known-problems/

**Comment by dsdrinnan** [comment]
Does someone know the specific kernel fix/change that resolved this in Linux kernel 3.18 or newer? I'm at a slightly older kernel, and don't currently have the option to update the kernel to 3.18+, but can request fixes.

**Comment by netblue30** [comment]
No, it has nothing do with OverlayFS. I think it was just on of those small kernel fixes that never get mentioned as a feature.
